frontend/src/v5/services/routing/routing.tsx

```diff
@@ -76,11 +82,58 @@ export const viewerRoute = (
 };
 
 interface RouteProps {
+	computedMatch?: any;
+	title?: string;
 	path?: string;
 	exact?: boolean;
 	children?: any;
 }
 
+const DEFAULT_TITLE = formatMessage({ id: 'pageTitle.default', defaultMessage: '3D Repo | Online BIM collaboration platform' });
+const LOADING_TEXT = formatMessage({ id: 'pageTitle.loading', defaultMessage: 'Loading...' });
+
+const truncateValue = (str, max = 12) => {
+	if (!str) return null;
+	let truncatedStr = str.substring(0, max).trim();
+	if (str.length > truncatedStr.length) truncatedStr += '...';
+	return truncatedStr;
+};
+
+export const Route = ({ title, children, computedMatch: { params }, ...props }: RouteProps) => {
+	const projectName = ProjectsHooksSelectors.selectCurrentProjectName();
+	const containerName = ContainersHooksSelectors.selectContainerById(params.containerOrFederation)?.name;
+	const federationName = FederationsHooksSelectors.selectFederationById(params.containerOrFederation)?.name;
+	const containerOrFederationName = containerName || federationName || '';
+	const revisionTag = useSelector(selectCurrentRevision)?.tag || LOADING_TEXT;
+
+	const PARAM_REGEX = /:(\S+)/g;
+	const parseTitle = (string) => string.replace(PARAM_REGEX, (_, match) => {
```

Comment:
I think this can be defined outside the component. It looks like a helper function.

Reply:
I wasn't able to take it outside the component in a neat way since it relies on things like `projectName` and `revisionTag` which come from hooks. I have tidied it up a bit though